### ISSUE TITLE
ERROR: Invalid UserOp signature or paymaster signature

### ISSUE THREAD

**Issue by 1329576606** [alchemyplatform/aa-sdk#43]
### [REQUIRED] Environment
Details: Invalid UserOp signature or paymaster signature
Version: viem@0.3.17
    at Object.http (/Users/user/eth/aa-account/node_modules/@alchemy/aa-core/node_modules/viem/src/utils/rpc.ts:110:13)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async request (/Users/user/eth/aa-account/node_modules/@alchemy/aa-core/node_modules/viem/src/clients/transports/http.ts:56:30)
    at async delay.count.count (/Users/user/eth/aa-account/node_modules/@alchemy/aa-core/node_modules/viem/src/utils/buildRequest.ts:68:18)
    at async attemptRetry (/Users/user/eth/aa-account/node_modules/@alchemy/aa-core/node_modules/viem/src/utils/promise/withRetry.ts:34:22) {
  details: 'Invalid UserOp signature or paymaster signature',
  docsPath: undefined,
  metaMessages: [
    'URL: https://polygon-mumbai.g.alchemy.com/v2/DH4n-QvjWVu44OvUFRtIOAaOvzsu1U9B',
    'Request body: {"method":"eth_sendUserOperation","params":[{"initCode":"0x","sender":"0xFAF9c42cd05063fe22A966a0886BD48e51Eef9F1","nonce":"0x3","callData":"0xb61d27f6000000000000000000000000490dd3ccc7ff94823d4f10bb6d97c7802957e6bf00000000000000000000000000000000000000000000000000038d7ea4c6800000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000000","signature":"0x92569ae9ecab9058307e464a68edd1e79cc8f1a9fca10404a4f59ab4a5120e2a0d50e5ee4ac04d54b74885d47b87f822cb005d29ef42dd3cce122a9b47f6cc971b","paymasterAndData":"0x","callGasLimit":"0x3400","verificationGasLimit":"0xa1c4","preVerificationGas":"0xab90","maxFeePerGas":"0xdbd76b5e","maxPriorityFeePerGas":"0xdbd76b3e"},"0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789"]}'
  ],
  shortMessage: 'RPC Request failed.',
  version: 'viem@0.3.17',
  cause: {
    code: -32507,
    message: 'Invalid UserOp signature or paymaster signature'
  },
  code: -32507
}
- Browser version: ts-node
- AA SDK version: last
- Package:

**Comment by dancoombs** [comment]
We need more info to help out here, can you attach some code samples or a way to reproduce the bug? This issue may occur if something changes in the user operation after the signature is calculated.

**Comment by 1329576606** [comment]
```ts
import {
  SimpleSmartContractAccount,
  SmartAccountProvider,
  type SimpleSmartAccountOwner,
} from "@alchemy/aa-core";
import { privateKeyToAccount, mnemonicToAccount } from "viem/accounts";
import { polygonMumbai } from "viem/chains";
import { toHex } from "viem";
import { PRIVATE_KEY } from "./privateKey";

async function main() {
  const SIMPLE_ACCOUNT_FACTORY_ADDRESS =
    "0x9406Cc6185a346906296840746125a0E44976454";

  // 1. define the EOA owner of the Smart Account
  // This is just one exapmle of how to interact with EOAs, feel free to use any other interface
  const ownerAccount = privateKeyToAccount(
    PRIVATE_KEY
  );
  // All that is important for defining an owner is that it provides a `signMessage` and `getAddress` function
  const owner: SimpleSmartAccountOwner = {
    // this should sign a message according to ERC-191
    signMessage: async (msg) =>
      ownerAccount.signMessage({
        message: toHex(msg),
      }),
    getAddress: async () => ownerAccount.address,
  };

  // 2. initialize the provider and connect it to the account
  const provider = new SmartAccountProvider(
    // the demo key below is public and rate-limited, it's better to create a new one
    // you can get started with a free account @ https://www.alchemy.com/
    "https://polygon-mumbai.g.alchemy.com/v2/DH4n-QvjWVu44OvUFRtIOAaOvzsu1U9B", // rpcUrl
    "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789", // entryPointAddress
    polygonMumbai // chain
  ).connect(
    (rpcClient) =>
      new SimpleSmartContractAccount({
        entryPointAddress: "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789",
        chain: polygonMumbai,
        factoryAddress: SIMPLE_ACCOUNT_FACTORY_ADDRESS,
        rpcClient,
        owner,
        // optionally if you already know the account's address
        // accountAddress: "0xFAF9c42cd05063fe22A966a0886BD48e51Eef9F1",
      })
  );
  // 3. send a UserOperation
    const { hash } = await provider.sendUserOperation({
      target: "0x490Dd3cCc7ff94823d4F10bb6d97c7802957e6bf",
      data: "0x",
      value: 1000000000000000n, // value: bigint or undefined
    });

    console.log(hash);
    
}
main();
```
It worked when I used @alchemy/aa-core@v0.1.0-alpha.0. It got an error when I used other versions

**Comment by dancoombs** [comment]
The fix for now is to modify 

```
message: toHex(msg),
```

to be: 

```
message: { raw: toHex(msg) },
```

We're working on a longer term fix, and updates to our example code, in https://github.com/alchemyplatform/aa-sdk/pull/42

**Comment by moldy530** [comment]
I've update the README and merged #42 . this should be working now. Make sure to update the latest version of `viem` as well (at least 1.1.7 is required)
